osf/models/sanctions.py

```diff
@@ -308,16 +312,20 @@ def _email_template_context(self, user, node, is_authorizer=False):
         return {}
 
     def _notify_authorizer(self, authorizer, node):
+        if self.should_suppress_emails:
```

Comment:
Can we just make this function conditional in the ask method? Seems like bad practice enter this function only to exit immediately.

Reply:
I agree - I was hesitant because it requires either moving `should_suppress_emails` from `EmailApprovableSanction` to `TokenApprovableSanction` (which doesn't fit logically) or moving `ask` from `TokenApprovableSanction` to `EmailApprovableSanction` which makes sense but required some adjustments to testing method calls. I decided to move `ask` and make the changes and think the code is more logical as a result.